sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/executeImmediate.scala

```diff
@@ -0,0 +1,186 @@
+          plan
+        } else if (posNodes.nonEmpty && namedNodes.nonEmpty) {
+          throw QueryCompilationErrors.invalidQueryMixedQueryParameters()
+        } else {
+          if (posNodes.nonEmpty) {
+            PosParameterizedQuery(
+              plan,
+              // We need to resolve arguments before Resolution batch to make sure
+              // that some rule does not accidentally resolve our parameters.
+              // We do not want this as they can resolve some unsupported parameters
+              resolveArguments(expressions))
+          } else {
+            val aliases = expressions.collect {
+              case (e: Alias) => e
+            }
+            val nonAliases = expressions.filter(!_.isInstanceOf[Alias])
+
+            if (nonAliases.nonEmpty) {
+              throw QueryCompilationErrors.invalidQueryAllParametersMustBeNamed(nonAliases)
+            }
+
+            NameParameterizedQuery(
+              plan,
+              aliases.map(_.name),
+              // We need to resolve arguments before Resolution batch to make sure
+              // that some rule does not accidentally resolve our parameters.
+              // We do not want this as they can resolve some unsupported parameters.
+              resolveArguments(aliases))
+          }
+        }
+
+        if (targetVariables.nonEmpty) {
+          SetVariable(targetVariables, queryPlan)
+        } else { queryPlan }
+    }
+
+  private def parseStatement(
+      queryString: String,
+      targetVariables: Seq[Expression]): LogicalPlan = {
+    // If targetVariables is defined, statement needs to be a query.
+    // Otherwise, it can be anything.
+    val plan = if (targetVariables.nonEmpty) {
+      try {
+        catalogManager.v1SessionCatalog.parser.parseQuery(queryString)
+      } catch {
+        case e: ParseException =>
+          // Since we do not have a way of telling that parseQuery failed because of
+          // actual parsing error or because statement was passed where query was expected,
+          // we need to make sure that parsePlan wouldn't throw
+          catalogManager.v1SessionCatalog.parser.parsePlan(queryString)
+
+          // Plan was successfully parsed, but query wasn't - throw.
+          throw QueryCompilationErrors.invalidStatementForExecuteInto(queryString)
+      }
+    } else {
+      catalogManager.v1SessionCatalog.parser.parsePlan(queryString)
+    }
+
+    // do not allow nested execute immediate
+    if (plan.containsPattern(EXECUTE_IMMEDIATE)) {
+      throw QueryCompilationErrors.nestedExecuteImmediate(queryString)
+    }
+
+    plan
+  }
+
+  private def getVariableReference(expr: Expression, nameParts: Seq[String]): VariableReference = {
+    lookupVariable(nameParts) match {
+      case Some(variable) => variable.copy(canFold = false)
+      case _ =>
+        throw QueryCompilationErrors
+          .unresolvedVariableError(nameParts, Seq("SYSTEM", "SESSION"), expr.origin)
```

Comment:
Could you use `CatalogManager.SYSTEM_CATALOG_NAME` and `CatalogManager.SESSION_NAMESPACE`